lib/lsif_graph.dart

```diff
@@ -187,38 +207,80 @@ class Metadata extends Element {
 
   String projectRoot;
 
-  Metadata(this.projectRoot) : super(id: 'meta');
+  Metadata(this.projectRoot);
 
   @override
   Map<String, Object> toLsif() => {
         ...super.toLsif(),
         'projectRoot': projectRoot,
-        'version': '0.5.0',
+        'version': '0.4.3',
         'positionEncoding': 'utf-16',
         'toolInfo': toolInfo,
       };
 
   Map<String, Object> get toolInfo => {
-        'name': 'simple_lsif',
+        'name': 'lsif_indexer',
         'args': [],
         'version': 'dev',
       };
 }
 
-class Contains extends Edge {
+/// A "contains" edge specifically for a [Document], which contains references and declarations.
+class DocumentContains extends Edge {
   @override
   String get label => 'contains';
 
-  Contains(this.container);
+  DocumentContains(this.container);
 
   Document container;
 
   @override
   Map<String, Object> toLsif() =>
       {...super.toLsif(), 'outV': container.jsonId, 'inVs': incomingEdges};
 
-  List<String> get incomingEdges => [
-        ...container.references,
-        ...container.declarations,
-      ].map((each) => each.range.jsonId).toList();
+  // TODO: Tidy this up, and the semantics of when things are added to declarations. See
+  // emitReferencesAndDeclarations in Document, where we push things into the references
+  // from declarations, but not vice versa.
+  List<String> get incomingEdges => {
+        for (var ref in container.references) ref.range.jsonId,
+        for (var ref in container.externalReferences) ref.range.jsonId,
+        for (var declaration in container.declarations)
+          declaration.range.jsonId,
+        for (var ref in container.references) ref.declaration.range.jsonId,
+      }.toList();
+}
+
+/// A "contains" edge specifically for a [Project], which contains [Document]s
+class ProjectContains extends Edge {
+  @override
+  String get label => 'contains';
+
+  ProjectContains(this.project);
+
+  Project project;
+
+  @override
+  Map<String, Object> toLsif() =>
+      {...super.toLsif(), 'outV': project.jsonId, 'inVs': incomingEdges};
+
+  List<String> get incomingEdges =>
+      project.nonEmptyDocuments.map((each) => each.jsonId).toList();
+}
+
+/// A comment, to make reading the file easier.
+///
+/// This is ignored by Sourcegraph. Note that lsif-validate will complain
+/// about these as unreachable, but that can be ignored.
+class Comment extends Element {
```

Comment:
Great idea to include these in the LSIF file! Seems like a really good way to increase the readability